planner/core/preprocess.go

```diff
@@ -815,6 +815,22 @@ func (p *preprocessor) checkCreateViewWithSelect(stmt ast.Node) {
 			s.LockInfo.LockType = ast.SelectLockNone
 			return
 		}
+
+		if s.From != nil {
+			tblNames := extractTableList(s.From.TableRefs, nil, false)
+			for _, tn := range tblNames {
+				tbl, err := p.tableByName(tn)
+				if err != nil {
+					p.err = err
+					return
+				}
+
+				if tbl.Meta().TempTableType == model.TempTableLocal {
```

Comment:
`tbl.Meta().TempTableType != model.TempTableNone` ?

Reply:
I don't think it is necessary to forbid global temporary table in view, and oracle also supports global temporary table in view definition:

Views can be created against temporary tables and combinations of temporary and permanent tables:

https://oracle-base.com/articles/misc/temporary-tables#miscellaneous